### PR TITLE
Add CHANGELOG.md entry for v1.2.0

### LINKED ISSUE #52: Is the time to release a new version for this project?
After previous review, there were long time passed. I think we should find the time to release a new version. Besides, should we discuss the next plan for this project ?

@Bplotka @brancz

### PR DESCRIPTION
**Pull request by brancz**
Closes #52 #43 

@Bplotka

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+* Allow configuring counters with `prometheus.CounterOpts`.
+
+### Changed
+
+* Remove usage of deprecated `grpc.Code()`.
+* Remove usage of deprecated `grpc.Errorf` and replace with `status.Errorf`.
+
+### Removed
+
+n/a
```
Comment (Bplotka) [inline CHANGELOG.md]: n/a? What do you mean? That we don't have changelog for below and 1.1.0 or what?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.2.0] - 2018-06-04
```
Comment (Bplotka) [inline CHANGELOG.md]: worth to add actual link to (non-existing yet) 1.2.0 release on github (https://github.com/grpc-ecosystem/go-grpc-prometheus/releases/tag/v1.2)

File: CHANGELOG.md (hunk 1 above)
Comment (Bplotka) [reply CHANGELOG.md]: if that is the case, can we actually put 1.1 and 1.0 headers with n/a mention (or more verbose version "changelog started from 1.2")

Comment (brancz) [reply CHANGELOG.md]: n/a stands for not applicable, because there was nothing removed in this release

File: CHANGELOG.md (hunk 2 above)
Comment (fengzixu) [reply CHANGELOG.md]: I agree this comment.

**Pull request merged**